### ISSUE TITLE
Grails 3.0.4: loop on startup when root(ERROR, ['STDOUT']) is commented

### ISSUE THREAD

**Issue by morgan134** [grails/grails-core#9153]
In logback.groovy, if we comment root:

`//root(ERROR, ['STDOUT'])`

We a loop with a Stackoverflow exception:

`Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'serverProperties' defined in class path resource [org/springframework/boot/autoconfigure/web/ServerPropertiesAutoConfiguration.class]: Initialization of bean failed; nested exception is java.lang.StackOverflowError`

`Caused by: java.lang.StackOverflowError
        at org.grails.config.NavigableMap.hashCode(NavigableMap.groovy)
        at java.util.Objects.hashCode(Objects.java:98)
        at java.util.HashMap$Node.hashCode(HashMap.java:296)
        at java.util.AbstractMap.hashCode(AbstractMap.java:507)
        at org.codehaus.groovy.util.HashCodeHelper.updateHash(HashCodeHelper.java:64)
        at org.grails.config.NavigableMap.hashCode(NavigableMap.groovy)
        at org.codehaus.groovy.util.HashCodeHelper.updateHash(HashCodeHelper.java:64)
        at org.grails.config.NavigableMap.hashCode(NavigableMap.groovy)
        at java.util.Objects.hashCode(Objects.java:98)
        at java.util.HashMap$Node.hashCode(HashMap.java:296)
        at java.util.AbstractMap.hashCode(AbstractMap.java:507)
        at org.codehaus.groovy.util.HashCodeHelper.updateHash(HashCodeHelper.java:64)
        at org.grails.config.NavigableMap.hashCode(NavigableMap.groovy)
        at org.codehaus.groovy.util.HashCodeHelper.updateHash(HashCodeHelper.java:64)
        at org.grails.config.NavigableMap.hashCode(NavigableMap.groovy)`

Version:

`| Grails Version: 3.0.4`
`| Groovy Version: 2.4.4`
`| JVM Version: 1.8.0_51`

**Comment by graemerocher** [comment]
Duplicates #9146

**Comment by nobeans** [comment]
Hi @graemerocher ,

Is it really the same issue as #9146? That seems to be caused by `EnvironmentDecryptApplicationInitializer` of spring-cloud. But this isn't so. This can be easily reproduced by the created app as follows:

```
grails create-app sample
cd sample
vi grails-app/conf/logback.groovy
// Comment out the `root(ERROR, ['STDOUT'])`
grails run-app
```

It's not serious problem because it's just a corner case and usually you don't need to do so.
Anyway, this and #9146 don't seem the same issue at least on the surface, though there might be a same fundamental cause.

**Comment by graemerocher** [comment]
@nobeans It is the same in that both cases result in `hashCode()` being called on `NavigableMap` which contains the bug that produces the stack overflow

**Comment by nobeans** [comment]
Sorry, I saw only the advice of workaround for `EnvironmentDecryptApplicationInitializer` and I overlooked that you'd already fixed `NavigableMap` for #9146.
https://github.com/grails/grails-core/commit/5fdf2b1a6dcc74f90cf9c4de80ece1379c05fedd

No problem. Thanks.
